contributors/design-proposals/security-model.md

```diff
@@ -0,0 +1,190 @@
+query an ACL'ed label. This approach is both complex and insufficient.
+
+1. The accidental omission of a label means that the system fails open.
+2. It becomes increasingly difficult to reason about permissions, especially if
+   set-based label selectors are used.
+3. Can ACLs be created on existing labels?
+   1. Yes: clients can be locked out of their workloads.
+   2. No: workloads must be restarted to change permissions.
+
+This approach also perturbs the existing behavior of label selectors. Does the
+"list pod" permission allow me to see pods with contentious ACLs? What about
+"list secrets"?
+
+Finally, this approach is insufficent, because not all relationships are defined
+by label selectors. Notably, secrets cannot be protected by label ACLs.
+
+## Explicit "ownership" (recommended)
+
+Explicit owners can be used to limit access to both objects and relationships. A
+service would reject both unauthorized changes to its body, and unauthorized
+pods as backends.
+
+The simplicity and correctness of this method recommend it.
+
+### Namespace as "owner"
+
+By scoping method permissions to namespaces, a crude ownership model can be
+constructed (all objects are owned by a namespace, and object-object
+relationships must have the same owner). The advantage of this approach is that
+it's possible today.
+
+There are also disadvantages to this method.
+
+1. Not all resources are namespaced (e.g., nodes and volumes)
+2. The myriad escalations and CIA failures make it harder to [separate
+   privileges](https://en.wikipedia.org/wiki/Privilege_separation)
+   within a namespace.
+3. The mapping between namespace and true ownership must be enforced by the
+   cluster admin.
+
+### Intentional ownership
+
+Rather than overloading namespace, Kubernetes can introduce an explicit concept
+of ownership. For example, a service owned by group\* "foo" could only be updated by members
+of "foo". The service's backends would also be limited to pods owned by a group.
+
+1. All resources would be owned.
+2. Scoping down permissions is simple with control over both object and relationship access\*\*.
+3. These owners could derive from an identity management system, so the onus of
+enforcement isn't on the cluster admin.
+
+A true ownership model solves the disadvantages of namespace-based ownership.
+But it's not clear that the improvements are worth the effort of implementation.
+
+\* Group is only used as an example. The system could be built to support
+users, roles, service accounts, or any other identifiable entity.
+
+\*\* It's also reasonable to control relationships with a reflective or membership
+constraint. This simplifies permissions management at the expense of
+flexibility, and gives no real improvement over problem (2) with namespaces.
+
+# Workloads
+
+The other face of Kubernetes exposed to customers is the workload environment.
+This section focuses on protecting workloads from the environment, and the
+environment from workloads.
+
```

Comment:
I think a section of container runtime isolation should also be added, e.g. via hypervisor-based runtimes (refer [frakti](https://github.com/kubernetes/frakti)). 

Hypervisors could provide a more consolidate isolation boundary than namespace-based runtimes. And it could ensure the minimal attack surface in multi-tenant environments.